### PR TITLE
make repo more generic

### PR DESCRIPTION
**Pull request by joejulian**
Make this repo more generic in order to support other repos.

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]

File: run-tests.sh [hunk 1]
```diff
@@ -2,39 +2,37 @@
...
+printf "testing %s branch %s\n" "${KUBEADDONS_REPO}" "${TESTING_BRANCH}"
 
 if [ -d "./addons" ]
 then
-  printf "Running inside the kubeaddons-enterprise repository"
-  KUBEADDONS_ENTERPRISE_PATH="."
-  KUBEADDONS_ENTERPRISE_TESTS_PATH="kubeaddons-tests/tests/kubeaddons-enterprise"
-  KUTTL_TEST_CONFIGURATION_PATH="kubeaddons-tests/kuttl-test.yaml"
-elif [ -d "$KUBEADDONS_ENTERPRISE_PATH" ]
+  printf "Running inside the ${KUBEADDONS_REPO} repository"
```
Comment (zmalik) [inline run-tests.sh]:
```suggestion
  printf "Running inside the %s repository\n" "${KUBEADDONS_REPO}"
```

File: run-tests.sh [hunk 2]
```diff
@@ -2,39 +2,37 @@
...
+  KUBEADDONS_REPO="."
```
Comment (zmalik) [inline run-tests.sh:31]:
```suggestion
  KUBEADDONS_TESTS_PATH="./kubeaddons-tests/tests/${KUBEADDONS_REPO}"
  KUBEADDONS_REPO="."
```
`KUBEADDONS_TESTS_PATH` should be set also to run tests from the addons repository, like `kubeaddons-enterprise` before we were setting it to `kubeaddons-tests/tests/kubeaddons-enterprise`

**Review by zmalik — CHANGES_REQUESTED** [review]
thank you @joejulian ! looking forward to seeing more tests here

I just tested the repository, running the test from the `kubeaddons-enterprise` repo and added some suggestions to fix the script for that use-case

File: run-tests.sh (hunk 2 above)
Comment (joejulian) [reply run-tests.sh:31]: We can't have tests in the repository or it breaks the old version of catalog that's in the wild.

Comment (zmalik) [reply run-tests.sh:31]: we don't have tests in the repository but it checks out this repository

**Review by zmalik — CHANGES_REQUESTED** [review]
when running these tests from `kubeaddons-enterprise` it fails for me
```
~/repos/kubeaddons-enterprise on  staging! ⌚ 18:46:05
$ (⎈ N/A:N/A) make kind-test     
bin/kind create cluster --wait 10s --image=kindest/node:v1.17.5
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.17.5) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
 ✗ Waiting ≤ 10s for control-plane = Ready ⏳ 
 • WARNING: Timed out waiting for Ready ⚠️
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/
kubeaddons-tests/run-tests.sh
testing kubeaddons-enterprise branch dev
Running inside the kubeaddons-enterprise repository
addon directory detected: ./addons/zookeeper/0.x
missing tests for addon directory kubeaddons-tests/tests/./addons/zookeeper/0.x
make: *** [kind-test] Error 1
```

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,10 @@
-KUTTL_VERSION=0.5.0
-KIND_VERSION=0.8.1
+KUBEADDONS_REPO ?= kubeaddons-enterprise
+TESTING_BRANCH ?= dev
+
+KIND_VERSION ?= 0.8.1
+KUBECONFIG ?= kubeconfig
```
Comment (zmalik) [inline Makefile]:
```suggestion
KUBECONFIG = kubeconfig
```
let's not do this, unless we are testing outside the KIND cluster as for today. Can you rename it to `KUBEADDONS_TEST_KUBECONFIG`?

Comment (joejulian) [reply Makefile]: reworked it to avoid shooting myself in the foot again (I deleted my own kubeconfig)

**Comment by zmalik** [comment]
@joejulian I got this error now when running from `kubeaddons-enterprise` 
```
~/repos/kubeaddons-enterprise on  staging! ⌚ 16:45:50
$ (⎈ N/A:N/A) make kind-test     
bin/kind create cluster --wait 10s --image=kindest/node:v1.17.5
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.17.5) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
 ✗ Waiting ≤ 10s for control-plane = Ready ⏳ 
 • WARNING: Timed out waiting for Ready ⚠️
Set kubectl context to "kind-kind"
You can now use your cluster with:
kubectl cluster-info --context kind-kind
Thanks for using kind! 😊
kubeaddons-tests/run-tests.sh
testing kubeaddons-enterprise branch dev
Running inside the kubeaddons-enterprise repository
addon directory detected: ./addons/zookeeper/0.x
addon directory detected: ./addons/kafka/1.x
addon directory detected: ./addons/cassandra/0.x
addon directory detected: ./addons/cassandra/1.x
addon directory detected: ./addons/jenkins/1.x
addon directory detected: ./addons/spark/1.x
Path /Users/zain/repos/kubeaddons-enterprise
running tests in kubeaddons-tests/tests/kubeaddons-enterprise/./addons/zookeeper/0.x
=== RUN   kuttl
    kuttl: harness.go:382: starting setup
    kuttl: harness.go:229: running tests using configured kubeconfig.
W0820 15:54:30.910174   12369 loader.go:223] Config not found: kubeconfig
    kuttl: harness.go:427: cleaning up
    kuttl: harness.go:502: fatal error getting client: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
--- FAIL: kuttl (0.00s)
FAIL
make: *** [kind-test] Error 1
```

**Review by zmalik — APPROVED** [review]
LGTM!
and it works with small change in kubeaddons-enterprise 
Thank you @joejulian

**Pull request merged**